No `BytesMut::zeroed` found in bytes 1.1

**Describe the bug**

Got an error while working on other stuff in DataFusion:

```
error[E0599]: no function or associated item named `zeroed` found for struct `BytesMut` in the current scope
  --> datafusion/core/src/physical_plan/file_format/chunked_store.rs:88:56
   |
88 | ...                   let mut buffer = BytesMut::zeroed(chunk_size);
   |                                                  ^^^^^^ function or associated item not found in `BytesMut`

```

I don't see `BytesMut.zeroed` in bytes 1.1: https://docs.rs/bytes/1.1.0/bytes/struct.BytesMut.html.
But it's there since 1.2: https://docs.rs/bytes/1.2.0/bytes/struct.BytesMut.html.

I upgraded `bytes` to 1.4 locally to pass build.

**To Reproduce**
Steps to reproduce the behavior:

**Expected behavior**
A clear and concise description of what you expected to happen.

**Additional context**
Add any other context about the problem here.

--- a/datafusion/core/Cargo.toml
+++ b/datafusion/core/Cargo.toml
@@ -63,7 +63,7 @@ apache-avro = { version = "0.14", optional = true }
 arrow = { version = "34.0.0", features = ["prettyprint"] }
 async-compression = { version = "0.3.14", features = ["bzip2", "gzip", "xz", "zstd", "futures-io", "tokio"], optional = true }
 async-trait = "0.1.41"
-bytes = "1.1"
+bytes = "1.4"
 bzip2 = { version = "0.4.3", optional = true }
 chrono = { version = "0.4.23", default-features = false }
 dashmap = "5.4.0"

